project/src/main/java/com/google/sps/Models/Database.java

```diff
@@ -189,21 +223,31 @@ public static boolean shareDocument(String hash, String email) {
     }
 
     addDocumentForUser(hash, userID);
-    addUserForDocument(hash, userID);
+    addUserForDocument(hash, userID, permissions);
 
     return true;
   }
 
   // Takes a Document hash and a userID
   // Adds the userID to the Document's list of users
-  public static void addUserForDocument(String hash, long userID) {
+  public static void addUserForDocument(String hash, long userID, String permissions) {
     Query query = new Query("Document").addFilter("hash", Query.FilterOperator.EQUAL, hash);
     Entity docEntity = getDatastore().prepare(query).asSingleEntity();
-    ArrayList<Long> userIDs = getDocumentUsers(hash);
 
-    userIDs.add(userID);
-    docEntity.setProperty("userIDs", userIDs);
-    getDatastore().put(docEntity);
+    if (permissions == "Editor") {
```

Comment:
maybe save this constant at the top of Database, eg
String EDITOR = "Editor";
...
if (permissions == EDITOR) {
}

so we don't run into small bugs like the wrong capitalization, etc

Reply:
got it!